src/libs/Localize/index.ts

```diff
@@ -45,8 +45,8 @@ function init() {
     }, {});
 }
 
-type PhraseParameters<T> = T extends (...args: infer A) => string ? A : never[];
-type Phrase<TKey extends TranslationPaths> = TranslationFlatObject[TKey] extends (...args: infer A) => unknown ? (...args: A) => string : string;
+type PhraseParameters<T> = T extends (arg: infer A) => string ? [A] : T extends (arg: infer A) => PluralFormPhase ? [A, number] : never[];
```

Comment:
Following the discussion from [this conversation](https://github.com/Expensify/App/pull/45892/files#r1696842532), I believe we need to handle the optional arguments within the PhraseParameters here. for not to add the undefined when argument is optional

```javascript
errors.zipPostCode = translate('privatePersonalDetails.error.incorrectZipFormat', undefined);
```

We could handle it like below, but I'm not sure if it's the best way. What are your suggestions on the approach @blazejkustra @shubham1206agra ?

```diff
- type PhraseParameters<T> = 
-    T extends (arg: infer A) => string ? [A] : 
-    T extends (arg: infer A) => PluralFormPhase ? [A, number] : 
-    never[];
+ type PhraseParameters<T> = 
+    T extends (arg?: infer A) => string ? [A?] :
+    T extends (arg: infer A) => PluralFormPhase ? [A, number]:
+    T extends (arg: infer A) => string ? [A] :
+    never[];
```